analyses/molecular-subtyping-MB/05-subtype-mb-shh.R

```diff
@@ -0,0 +1,300 @@
+    dkfz_v12_methylation_subclass_collapsed = if(all(is.na(dkfz_v12_methylation_subclass))) NA_character_ 
+    else paste(unique(na.omit(dkfz_v12_methylation_subclass)), collapse = "; "),
+    dkfz_v12_methylation_subclass_score_mean = mean(dkfz_v12_methylation_subclass_score, na.rm = TRUE),
+    .groups = "drop"
+  ) %>%
+  mutate(across(where(is.numeric), ~replace(., is.nan(.), NA)),
+         SHH_subtype = case_when(dkfz_v12_methylation_subclass_score_mean >= 0.7 & dkfz_v12_methylation_subclass_collapsed == "MB_SHH_1" ~ "SHH_beta",
+                                 dkfz_v12_methylation_subclass_score_mean >= 0.7 & dkfz_v12_methylation_subclass_collapsed == "MB_SHH_2" ~ "SHH_gamma",
+                                 dkfz_v12_methylation_subclass_score_mean >= 0.7 & dkfz_v12_methylation_subclass_collapsed == "MB_SHH_3" ~ "SHH_alpha",
+                                 dkfz_v12_methylation_subclass_score_mean >= 0.7 & dkfz_v12_methylation_subclass_collapsed == "MB_SHH_4" ~ "SHH_delta",
+                                 TRUE ~ NA_character_))
+
+mb_subtypes <- mb_shh %>%
+  select(match_id, molecular_subtype, molecular_subtype_methyl) %>%
+  unique()
+
+final_subtypes <- mb_subtypes %>%
+  left_join(hist_mb_shh) %>%
+  left_join(mb_shh_data_collapsed) %>%
+  left_join(germline) %>%
+  left_join(expr_df) %>%
+  unique() %>%
+  # clean up subclass and scores
+  mutate(dkfz_v12_methylation_subclass_score_mean = case_when(dkfz_v12_methylation_subclass_score_mean < 0.7 & 
+                                                                !grepl("SHH", dkfz_v12_methylation_subclass_collapsed) ~ NA_integer_,
+                                                              TRUE ~ dkfz_v12_methylation_subclass_score_mean),
+         dkfz_v12_methylation_subclass_collapsed = case_when((dkfz_v12_methylation_subclass_score_mean < 0.7 | is.na(dkfz_v12_methylation_subclass_score_mean)) & 
+                                                               !grepl("SHH", dkfz_v12_methylation_subclass_collapsed) ~ NA_character_,
+                                                             TRUE ~ dkfz_v12_methylation_subclass_collapsed)) %>%
+  mutate(final_shh_subtype = case_when(
+    # For SHH alpha: >3 years AND (MYCN OR GLI2 amplification OR TPM>=2 OR chr17p loss OR 9p gain with lower conf methyl)
+    SHH_subtype == "SHH_alpha" | (age_at_diagnosis_years >= 2 &
+                                    is.na(SHH_subtype) &
+                                    (grepl("amplification", consensus_CN_MYCN) |
+                                       grepl("amplification", consensus_CN_GLI2) |
+                                       MYCN_TPM_zscore >= 2 |
+                                       GLI2_TPM_zscore >= 2 |
+                                       `17p_loss` == 1 |
+                                       (`9p_gain` == 1 & dkfz_v12_methylation_subclass_collapsed == "MB_SHH_3" & 
+                                          dkfz_v12_methylation_subclass_score_mean >= 0.5)
```

Comment:
I added this, because I noticed this one was likely alpha, but can also go either way with leaving it out- asked Sebastian to take a look at these also for us.